### PR TITLE
ipc: simplify CMakeLists.txt by using a suffix variable

### PR DESCRIPTION
**Pull request by lyakh**
Everywhere, where IPC3 and IPC4 are distinguished in CMakeLists.txt the only difference is the use of x_ipc3.c vs. x_ipc4.c. Instead of duplicating all build instructions for IPC3 and IPC4 just define and use a suffix variable.

### CODE REVIEW

File: zephyr/CMakeLists.txt [hunk 1]
```diff
@@ -734,23 +694,18 @@ zephyr_library_sources_ifdef(CONFIG_COMP_CHAIN_DMA
...
-		${SOF_AUDIO_PATH}/src/src_ipc3.c
-	)
-elseif(CONFIG_IPC_MAJOR_4)
-	zephyr_library_sources_ifdef(CONFIG_COMP_SRC
+if(CONFIG_COMP_SRC STREQUAL "m")
+	add_subdirectory(${SOF_AUDIO_PATH}/src/llext
+			 ${PROJECT_BINARY_DIR}/src_llext)
+	add_dependencies(app src)
+elseif(CONFIG_COMP_SRC)
+	zephyr_library_sources(
```
Comment (kv2019i) [inline zephyr/CMakeLists.txt:702]: I think this is adding an unrelated change related to llext...

**Review by kv2019i — CHANGES_REQUESTED** [review]
Change looks good, but I think you hid an easter egg in the commit...

File: zephyr/CMakeLists.txt (hunk 1 above)
Comment (lyakh) [reply zephyr/CMakeLists.txt:702]: @kv2019i oh, just a tiny small one, wouldn't hurt! ;-) No, thanks for catching, indeed!

**Review by marc-hb — COMMENTED** [review]
Nice! I remember trying to stem that copy/paste/diverge tide when it happened but I wasn't successful.

One concern though: is there still some Zephyr+IPC3 platform that is actually running this? Some old IMX config maybe?

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by marc-hb** [comment]
SOFCI TEST

**Comment by lyakh** [comment]
> One concern though: is there still some Zephyr+IPC3 platform that is actually running this? Some old IMX config maybe?

@marc-hb I don't know but why is this a concern? Would this patch break such cases?

**Review by kv2019i — APPROVED** [review]

**Comment by marc-hb** [comment]
>  I don't know but why is this a concern? Would this patch break such cases?

Yes of course: if it's not tested then it does not work.

**Review by lgirdwood — APPROVED** [review]
More deletion than additions are good here !

**Comment by lgirdwood** [comment]
> > One concern though: is there still some Zephyr+IPC3 platform that is actually running this? Some old IMX config maybe?
> 
> @marc-hb I don't know but why is this a concern? Would this patch break such cases?

Build test only.

**Review by iuliana-prodan — APPROVED** [review]
LGTM

**Comment by iuliana-prodan** [comment]
> > > One concern though: is there still some Zephyr+IPC3 platform that is actually running this? Some old IMX config maybe?
> > 
> > 
> > @marc-hb I don't know but why is this a concern? Would this patch break such cases?
> 
> Build test only.

I haven't tested on target, but from code review and the SOF components we are using with i.MX the changes are ok.

**Comment by lyakh** [comment]
CI failures:
1. a known Zephyr "main" breakage
2. a known Zephyr sparse breakage
3. unrelated sof-logger breakage like https://sof-ci.01.org/sofpr/PR9180/build4990/devicetest/index.html?model=LNLM_RVP_NOCODEC&testcase=check-sof-logger similar to https://sof-ci.01.org/sofpr/PR9182/build5002/devicetest/index.html?model=LNLM_RVP_NOCODEC&testcase=check-sof-logger
4. some weird "kernel boot log flood" https://sof-ci.01.org/sofpr/PR9180/build4990/devicetest/index.html?model=LNLM_RVP_HDA&testcase=verify-kernel-boot-log
5. pause-resume failures like https://sof-ci.01.org/sofpr/PR9180/build4990/devicetest/index.html?model=LNLM_RVP_HDA&testcase=check-pause-resume-capture-100 similar to https://sof-ci.01.org/sofpr/PR9182/build5002/devicetest/index.html?model=LNLM_RVP_HDA&testcase=check-pause-resume-capture-100
6. alsabat failures shouldn't be related either
7. unrelated suspend-resume failures

In general, this is a build enhancement, so it is unexpected to introduce subtle run-time failures.

**Comment by lyakh** [comment]
> > I don't know but why is this a concern? Would this patch break such cases?
> 
> Yes of course: if it's not tested then it does not work.

@marc-hb sorry, not sure I'm getting your answer. My question was, whether **this** PR was breaking it, not whether it works.

**Comment by marc-hb** [comment]
This PR changes "if (IPC3/IPC4)" code in `zephyr/cmakelist.txt`. If nothing compiles any Zephyr+IPC3 configuration right now, then the IPC3 branch of the change is not tested.

I'm not 100% sure but I think @iuliana-prodan said there is such an active configuration right now.
